src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp

```diff
@@ -16,384 +16,217 @@ namespace DB
+        * done between client's exchange and local bridge exchange. Binding key must be a string integer in case of hash exchange, for
+        * fanout exchange it can be arbitrary
+        */
+        setup_channel->bindQueue(exchange_name, queue_name, std::to_string(channel_id_base))
+        .onSuccess([&]
+        {
+            binding_created = true;
+        })
+        .onError([&](const char * message)
         {
-            local_exchange_declared = false;
-            LOG_ERROR(log, "Failed to bind local direct-exchange to fanout-exchange. Reason: {}", message);
+            throw Exception("Failed to create queue binding. Reason: " + std::string(message), ErrorCodes::LOGICAL_ERROR);
         });
+    };
 
-        return;
-    }
-
-    AMQP::ExchangeType type;
-    if      (exchange_type == ExchangeType::FANOUT)         type = AMQP::ExchangeType::fanout;
-    else if (exchange_type == ExchangeType::DIRECT)         type = AMQP::ExchangeType::direct;
-    else if (exchange_type == ExchangeType::TOPIC)          type = AMQP::ExchangeType::topic;
-    else if (exchange_type == ExchangeType::HASH)           type = AMQP::ExchangeType::consistent_hash;
-    else if (exchange_type == ExchangeType::HEADERS)        type = AMQP::ExchangeType::headers;
-    else throw Exception("Invalid exchange type", ErrorCodes::BAD_ARGUMENTS);
-
-    /* Declare client's exchange of the specified type and bind it to hash-exchange (if it is not already hash-exchange), which
-     * will evenly distribute messages between all consumers. (This enables better scaling as without hash-exchange - the only
-     * option to avoid getting the same messages more than once - is having only one consumer with one queue)
-     */
-    consumer_channel->declareExchange(exchange_name, type).onError([&](const char * message)
+    auto error_callback([&](const char * message)
     {
-        local_exchange_declared = false;
-        LOG_ERROR(log, "Failed to declare client's {} exchange. Reason: {}", exchange_type, message);
+        /* This error is most likely a result of an attempt to declare queue with different settings if it was declared before. So for a
+         * given queue name either deadletter_exchange parameter changed or queue_size changed, i.e. table was declared with different
+         * max_block_size parameter. Solution: client should specify a different queue_base parameter or manually delete previously
+         * declared queues via any of the various cli tools.
+         */
+        throw Exception("Failed to declare queue. Probably queue settings are conflicting: max_block_size, deadletter_exchange. Attempt \
+                specifying differently those settings or use a different queue_base or manually delete previously declared queues,      \
+                which  were declared with the same names. ERROR reason: "
+                + std::string(message), ErrorCodes::LOGICAL_ERROR);
```

Comment:
Probably it's not a logical error? It depends on user input? If so `BAD_ARGUMENTS` would be preferable.